packages/lib/server/repository/organization.ts

```diff
@@ -0,0 +1,68 @@
+import { IS_TEAM_BILLING_ENABLED } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+import { safeStringify } from "@calcom/lib/safeStringify";
+import { prisma } from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
+
+import { createAProfileForAnExistingUser } from "../../createAProfileForAnExistingUser";
+
+export class OrganizationRepository {
+  static async createWithOwner({
```

Comment:
Part of my wonders if we'd be better putting this in a interactive transaction -> if creating the owner or membership fails i think it'd be better to rollback than having the floating org left?

Reply:
Yeah that is the best. But I was having trouble getting it working with repository pattern as that needs me to pass **tx** to repository methods or have some sort of duplication. 

e.g. Here I can reuse `createAProfileForAnExistingUser`. Either I start supporting `tx` argument with it or I copy the code for that here probably, so that it has access to `tx`

Reply:
Its not blocking and can deffo be done in a follow up